org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala

```diff
@@ -45,32 +46,45 @@ class SbtInputs(installation: IScalaInstallation,
     addToClasspath: Seq[IPath] = Seq.empty,
     srcOutputs: Seq[(IContainer, IContainer)] = Seq.empty) {
 
-  def cache = CompilerCache.fresh // May want to explore caching possibilities.
+  def cache = new FreshCompilerCache // May want to explore caching possibilities.
 
   private val allProjects = project +: project.transitiveDependencies.flatMap(ScalaPlugin().asScalaProject)
 
-  def analysisMap(f: File): Maybe[CompileAnalysis] =
+  def analysisMap(f: File): Optional[CompileAnalysis] =
     if (f.isFile)
-      Maybe.nothing[CompileAnalysis]
+      Optional.empty[CompileAnalysis]
     else {
       val analysis = allProjects.collectFirst {
         case project if project.buildManager.buildManagerOf(f).nonEmpty =>
           project.buildManager.buildManagerOf(f).get.latestAnalysis
       }
       analysis.map { analysis =>
-        Maybe.just(analysis)
-      }.getOrElse(Maybe.nothing[CompileAnalysis])
+        Optional.ofNullable(analysis)
+      }.getOrElse(Optional.empty[CompileAnalysis])
     }
 
-  def progress = Maybe.just(scalaProgress)
+  def progress = Optional.ofNullable(scalaProgress)
 
   def incOptions: IncOptions = {
-    IncOptionsUtil.defaultIncOptions().
+    import xsbti.compile.IncOptions._
+    val base = of()
+    base.
       withApiDebug(project.storage.getBoolean(SettingConverterUtil.convertNameToProperty(preferences.ScalaPluginSettings.apiDiff.name))).
       withRelationsDebug(project.storage.getBoolean(SettingConverterUtil.convertNameToProperty(preferences.ScalaPluginSettings.relationsDebug.name))).
-      withClassfileManagerType(Maybe.just(new TransactionalManagerType(tempDir, logger))).
-      withApiDumpDirectory(Maybe.nothing()).
-      withRecompileOnMacroDef(Maybe.just(project.storage.getBoolean(SettingConverterUtil.convertNameToProperty(preferences.ScalaPluginSettings.recompileOnMacroDef.name))))
+      withClassfileManagerType(Optional.ofNullable(TransactionalManagerType.create(tempDir, logger))).
+      withApiDumpDirectory(Optional.empty()).
+      withRecompileOnMacroDef(Optional.ofNullable(project.storage.getBoolean(SettingConverterUtil.convertNameToProperty(preferences.ScalaPluginSettings.recompileOnMacroDef.name)))).
+      withEnabled(defaultEnabled()).
+      withStoreApis(defaultStoreApis()).
+      withApiDiffContextSize(defaultApiDiffContextSize()).
+      withExternalHooks(defaultExternal()).
+      withExtra(defaultExtra()).
+      withLogRecompileOnMacro(defaultLogRecompileOnMacro()).
+      withRecompileAllFraction(defaultRecompileAllFraction()).
+      withRelationsDebug(defaultRelationsDebug()).
+      withTransitiveStep(defaultTransitiveStep()).
+      withUseCustomizedFileManager(defaultUseCustomizedFileManager()).
+      withUseOptimizedSealed(defaultUseOptimizedSealed())
```

Comment:
Are all of these `with...(default...)` calls required or could they just be left out?